src/cobra/core/reaction.py

```diff
@@ -177,29 +204,37 @@ def objective_coefficient(self):
         return linear_reaction_coefficients(self.model, [self]).get(self, 0)
 
     @objective_coefficient.setter
-    def objective_coefficient(self, value):
+    def objective_coefficient(self, value: float) -> None:
         if self.model is None:
             raise AttributeError("cannot assign objective to a missing model")
         if self.flux_expression is not None:
             set_objective(self.model, {self: value}, additive=True)
 
-    def __copy__(self):
+    def __copy__(self) -> "Reaction":
+        """Copy the Reaction."""
         cop = copy(super(Reaction, self))
         return cop
 
-    def __deepcopy__(self, memo):
+    # Unclear what memo should be
+    def __deepcopy__(self, memo) -> "Reaction":
```

Comment:
Same as above.

Reply:
Done. Can you help me figure out what memo should be?

Reply:
It's a `dict` https://docs.python.org/3/library/copy.html#copy.deepcopy

```suggestion
    def __deepcopy__(self, memo: dict) -> "Reaction":
```

Reply:
According to https://docs.python.org/3/library/copy.html , `memo` should be a dictionary, so the type can be `Dict`. A generic `Dict` without key and value type is fine in my opinion.